ocrd_utils/ocrd_utils/os.py

```diff
@@ -118,7 +118,7 @@ def get_processor_resource_types(executable, ocrd_tool=None):
         result = run([executable, '--dump-json'], stdout=PIPE, check=True, universal_newlines=True)
         ocrd_tool = loads(result.stdout)
     has_dirs = next((True for p in ocrd_tool['parameters'].values() if p.get('content-type', None) == 'text/directory'), False)
-    has_files = next((True for p in ocrd_tool['parameters'].values() if p.get('content-type', None) != 'text/directory'), False)
+    has_files = next((True for p in ocrd_tool['parameters'].values() if 'content-type' in p and p['content-type'] != 'text/directory'), False)
```

Comment:
Boolean logic is surprisingly hard at times :) But your proposal won't help either because the `format: uri` (or `format: file`) are not what is required. As the spec states

> To mark a parameter as expecting the address of a file, it must declare the content-type property as a valid media type.

`content-type` actually is the best indicator that this is a resource parameter. So no way around fixing the ocrd-tool.json of the processors AFAICT. Do you see another solution?

Reply:
E.g. in ocrd-tesserocr-recognize, `model` is just

```
  "model": {
   "type": "string",
   "description": "The tessdata text recognition model to apply (an ISO 639-3 language specification or some other basename, e.g. deu-frak or Fraktur)."
  },
```

(so we did not eat our own dog food and now it's coming back to bite us).

Reply:
834b4d16 I really want this done, so I added a workaround:

```
if not next((True for p in ocrd_tool['parameters'].values() if 'content-type' in p), False):
    # None of the parameters for this processor are resources (or not
    # the resource parametrs are not properly declared, so output both
    # directories and files
    return (True, True)
```

This should ensure that the current behavior of listing all the files/directories is retained if a processor does not define resource parameters. That should not be the case, because there should be no files then but better to be safe. Once we fix the ocrd-tool.json, the result will be properly filtered.

Reply:
Sounds like a reasonable compromise for now and way forward. And since it's already merged, I won't puzzle over its ramifications, but observe :-)